src/Controller/V2/Quarterback/QuarterbackSchema.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * © 2020 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Joi from 'joi'
+
+export const createDocumentSchema: Joi.SchemaMap = {
+  params: Joi.object({
+    projectID: Joi.string().required(),
+    manuscriptID: Joi.string().required(),
+  }),
+  body: Joi.object({
+    manuscript_model_id: Joi.object().required(),
+    project_model_id: Joi.object().required(),
+    doc: Joi.object().required(),
+  }),
+}
+export const updateDocumentSchema: Joi.SchemaMap = {
+  params: Joi.object({
+    projectID: Joi.string().required(),
+    manuscriptID: Joi.string().required(),
+  }),
+  body: Joi.object({
+    doc: Joi.object().required(),
+  }),
+}
+export const getDocumentSchema: Joi.SchemaMap = {
+  params: Joi.object({
+    projectID: Joi.string().required(),
+    manuscriptID: Joi.string().required(),
+  }),
+}
+export const deleteDocumentSchema: Joi.SchemaMap = {
+  params: Joi.object({
+    projectID: Joi.string().required(),
+    manuscriptID: Joi.string().required(),
+  }),
+}
+
+export const createSnapshotSchema: Joi.SchemaMap = {
+  params: Joi.object({
+    projectID: Joi.string().required(),
+    manuscriptID: Joi.string().required(),
+  }),
+  body: Joi.object({
+    name: Joi.string().required(),
+    docId: Joi.string().required(),
```

Comment:
can we call this `docID` for consistency?

Reply:
We can, but FE need to be aware that this param will change (I'm trying to simplify the migration on FE side as much as possible)

Reply:
Also this would either require remapping this param in manuscriptsAPI **docID** -> **docId**, or update quarterbackAPI to expect the param to be **docID** instead **docId**

Reply:
OK, so this is an existing param. Can we change quarterback to accept both so that you don't need to do the mapping in manuscripts-api? FE can switch to the new param name when they work on their part.

Reply:
It is a bit finicky with the way the schema of the endpoints are written in Quarterback, it would be easier to just move to the new naming ;docID' instead of supporting both